src/index.ts

```diff
@@ -105,6 +118,11 @@ player.volume(1)
 player.controls(true)
 
 getVideoSrc(query).then((src) => {
+
+  document.body.addEventListener("loadstart",function(e){
+    metrics.reportGenericVideoMetrics(e.target, reportingUrl);
```

Comment:
Try calling `reportGenericVideoMetrics` on the `video` element here as well.

That (and maybe this) should also go _after_ you configure the `src` of the player below, on the line that I had linked you in the end of this callback.